File: icebridge_python_worker/mdworker.py
"""Majordomo Protocol worker example.

Uses the mdwrk API to hide all MDP aspects

Author: Min RK <benjaminrk@gmail.com>
"""

import sys
from mdwrkapi import MajorDomoWorker
from group_manager import GroupManager


def main():
    verbose = '-v' in sys.argv
    worker = MajorDomoWorker("tcp://localhost:5555", "python_service", True)
    reply = None
    import os
    worker_root = os.path.dirname(os.path.abspath(__file__))
    print os.path.join(worker_root,"python_worker.conf")
    group_instace = GroupManager(config_file_path=os.path.join(worker_root,"python_worker.conf"))
    while True:
        request = worker.recv(reply)
        if request is None:
            break # Worker was interrupted
        ''' reply is a list, the messge that we want to return is [msg]
            the request we receive is [msg]
        '''
        reply = group_instace.process(request)


if __name__ == '__main__':
    main()

File: icebridge_svn/personal/xiong/logic_server_20131023/test_logic_workers/mdworker.cpp
//
//  Majordomo Protocol worker example
//  Uses the mdwrk API to hide all MDP aspects
//
//  Lets us 'build mdworker' and 'build all'
//
//     Andreas Hoelzlwimmer <andreas.hoelzlwimmer@fh-hagenberg.at>
//
#include "mdwrkapi.hpp"

int main (int argc, char *argv [])
{
     if (argc < 5) {
         printf ("syntax: worker worker_name broker_address server_type -v\n");
         return 0;
     }
     printf("%d\n" ,argc);
     char * identity = argv [1];
     char * broker_address = argv [2];
     char * server_type = argv [3];

    int verbose = (argc > 1 && strcmp (argv [4], "-v") == 0);
    //mdwrk session (identity, "tcp://localhost:5556", "echo", verbose);
    mdwrk session (identity, broker_address, "echo", verbose);

    zmsg *reply = 0;
    while (1) {
        zmsg *request = session.recv (reply);
        if (request == 0) {
            break;              //  Worker was interrupted
        }
        request->body_set("hehe\0hehe",9);
        reply = request;        //  Echo is complex... :-)
    }
    return 0;
}
